combine.lua

```diff
@@ -0,0 +1,306 @@
+-- Merge food and drink stacks in the selected stockpile or every stockpile
+--[====[
+
+combine by Vettlingr
+==============
+Merge stacks of food in the selected stockpile or all stockpiles.
+
+]====]
+local utils = require 'utils'
+
+local f = {
+    validArgs = utils.invert({ 'help', 'drinks', 'plants', 'meat', 'fish', 'fat', 'food', 'roasts', 'max', 'all', 'stockpile' });
+    args = utils.processArgs({...}, validArgs);
+help = [====[
+Combine
+=============
+Merge stacks of food in selected Stockpile or across all stockpiles on the map.
+Valid commands:
```

Comment:
There are some patterns from other scripts that we can follow here to make the API more consistent. How about:
combine -> displays help
combine here -> combines everything it can for the stockpile under the cursor
combine all -> combines everything it can for all stockpiles
combine all --types=plants,roasts --max=100 -> combines all plant and roast barrels with a custom max stack size

So the syntax would be:
```
combine [help|here|all] [<options>]

Where 'help' (or nothing) displays help, 'here' directs combine to act on the selected stockpile, and 'all' directs combine to act on all stockpiles.

Options can be:
-t, --types <comma separated list of types>
    Restricts combine to acting on the specified types. Valid types are: ...
-m, --max <number>
    Sets the preferred target stack size. If not specified, defaults to ...
 
```

The `argparse` module will help you support that syntax. The prioritize script is a good model to follow.
https://github.com/DFHack/scripts/blob/master/prioritize.lua#L532